fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1693,9 +1693,9 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
             if hasattr(param._linked_param, "_is_shared") and param._linked_param._is_shared:
                 param = param._linked_param
 
-        assert param.grad is not None, param.shape
-        if param.grad.requires_grad:
-            raise RuntimeError("FSDP only works with gradients that don't require gradients")
+        # assert param.grad is not None, param.shape
```

Comment:
Perhaps some check is needed to make sure parameters are not shared (as would be the case with weights tying)?